ansible_base/oauth2_provider/models/application.py

```diff
@@ -54,14 +56,25 @@ class Meta(oauth2_models.AbstractAccessToken.Meta):
         on_delete=models.CASCADE,
         null=True,
     )
-    # Not overriding client_secret... Details:
+
     # It would be nice to just use our usual encrypted_fields flow here
     # until DOT makes a release with https://github.com/jazzband/django-oauth-toolkit/pull/1311
     # there is no way to disable its expectation of using its own hashing
     # (which is Django's make_password/check_password).
     # So we use their field here.
     # Previous versions of DOT didn't hash the field at all and AWX pins
     # to <2.0.0 so AWX used the AWX encryption with no issue.
+    # We still override it here so that we can prevent_search() on it.
+    client_secret = prevent_search(
```

Comment:
I'm surprised this doesn't require/complain about a migration.

From 0001 migration.
`('client_secret', models.CharField(blank=True, db_index=True, default=oauth2_provider.generators.generate_client_secret, help_text='Used for more stringent verification of access to an application when creating a token.', max_length=1024)),`

Reply:
0002 modifies it some and restores it to the upstream defaults. And this change keeps all those parameters the same as upstream, just wrapping it in `prevent_search()` which IIRC sets a field property but not one that Django knows about.